File: tests/tests_old_compiler/ConditionalRemoteCalls/jt34.c

jtask* localfunc(int n, char *s) {
        
    while(1) {
        printf("Indx %d message - %s\n", n, s);
        jsleep(500000);
    }
}

jtask* {typeAonly} workerFuncX(char *s) {
    printf("WorkerFuncX -- %s\n", s);
}

jtask* {typeBonly} workerFuncY(char *s) {
    printf("WorkerFuncY -- %s\n", s);
}


int main(int argc, char *argv[])
{
    localfunc(10, "a-local-task");
    return 0;
}

File: tests/tests_old_compiler/ConditionalRemoteCalls/jt34.js

jcond {
    typeAonly: jsys.tag == "typeA";
    typeBonly: jsys.tag == "typeB";
}

setInterval(()=> {
    console.log("This is a loop at the controller.", jsys.tags);
    workerFuncX("hello, from controller").catch((e)=>{
        console.log("calling workerFuncX has an Error: " + e);
    });
    workerFuncY("hello, from controller").catch((e)=>{
        console.log("calling workerFuncY has an Error: " + e);
    });

}, 1000);